pkg/auth/auth.go

```diff
@@ -0,0 +1,240 @@
+package auth
```

Comment:
Here there is a lot of functionality which could be simplified by using an existing [module](github.com/coreos/go-oidc). It is fine as it is but I would have possibly opted for trying that out.

Reply:
Not sure what could be done better with that library. It seems to be an OIDC client, and DOA is not one.

The code there is from a JWT library and used only to parse, verify and validate JWTs.

Reply:
I think you replicate exactly what a oidc client can do in this code. It would be good to put this in a module or using an existing one.

Reply:
Solved. This service does not require the complete auth flow. Only a token.